### PR TITLE
Update continuous integration

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Make continuous integration work again.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-tac?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -130,3 +132,12 @@ def network_node(oef_addr, oef_port, pytestconfig):
             c.stop()
             c.remove()
             logger.info("Stopped the OEF node.")
+
+
+@pytest.fixture(scope="session", autouse=True)
+def environ():
+    """Set the environment."""
+    d = tempfile.mkdtemp()
+    os.environ["TAC_SHARED_DIR"] = d
```
Comment (DavidMinarsch) [inline tests/conftest.py:141]: Note, different name to above...

File: sandbox/docker-compose.yml [hunk 2]
```diff
@@ -47,8 +47,6 @@ services:
     depends_on:
       - oef-node
       - visdom
-    volumes:
```
Comment (DavidMinarsch) [inline sandbox/docker-compose.yml:50]: are you sure we do not need it anymore?

Comment (marcofavorito) [reply sandbox/docker-compose.yml:50]: the `volumes` section was duplicate, see below

File: sandbox/.env [hunk 3]
```diff
@@ -17,4 +17,4 @@ WHITELIST=
 REGISTER_AS=both
 SEARCH_FOR=both
 PENDING_TRANSACTION_TIMEOUT=120
-SHARED_DIR=/agents-tac/data/shared
\ No newline at end of file
+SHARED_DIR=../data/shared
```
Comment (marcofavorito) [inline sandbox/.env:20]: this must be relative to `sandbox/`

File: .github/workflows/workflow.yaml [hunk 4]
```diff
@@ -17,7 +17,7 @@ jobs:
        - uses: actions/checkout@master
        - uses: actions/setup-python@master
          with:
-           python-version: 3.6
+           python-version: 3.7
```
Comment (marcofavorito) [inline .github/workflows/workflow.yaml:20]: Python 3.6 does not work with `oef`

File: tac/agents/participant/v1/agent.py [hunk 5]
```diff
@@ -145,6 +145,7 @@ def start(self, rejoin: bool = False) -> None:
             self.oef_handler.rejoin = rejoin
             super().start()
             self.oef_handler.rejoin = False
+            self.stop()
```
Comment (marcofavorito) [inline tac/agents/participant/v1/agent.py:148]: baseline agent v1 didn't stop the execution with `self.stop`, but setting `self.liveness._is_stopped` to `True`. However, that makes `mailbox.disconnect()` never being called.

File: .github/workflows/workflow.yaml [hunk 6]
```diff
@@ -65,4 +65,4 @@ jobs:
         brew install protobuf
     - name: Unit tests
       run: |
-        tox -e py${{ matrix.python-version }}
+        tox -e py${{ matrix.python-version }} -- --ci --no-oef
```
Comment (DavidMinarsch) [inline .github/workflows/workflow.yaml:68]: We should try running all of them

**Pull request merged**